### ISSUE TITLE
ARM64/Linux: CMake initialization failure on ARM64 cross build

### ISSUE THREAD

**Issue by lemmaa** [dotnet/coreclr#5405]
same case with #5316
build.sh cross arm64 clean verbose fails with the following error message:

```
-- Performing Test SEM_INIT_MODIFIES_ERRNO
-- Performing Test SEM_INIT_MODIFIES_ERRNO - Failed
-- Performing Test HAVE_PROCFS_CTL
CMake Error: TRY_RUN() invoked in cross-compiling mode, please set the following cache variables appropriately:
   HAVE_PROCFS_CTL_EXITCODE (advanced)
For details see /home/sjlee/git/coreclr/bin/obj/Linux.arm64.Debug/TryRunResults.cmake
-- Performing Test HAVE_PROCFS_CTL - Failed
-- Performing Test HAVE_PROCFS_MAPS
CMake Error: TRY_RUN() invoked in cross-compiling mode, please set the following cache variables appropriately:
   HAVE_PROCFS_MAPS_EXITCODE (advanced)
For details see /home/sjlee/git/coreclr/bin/obj/Linux.arm64.Debug/TryRunResults.cmake
-- Performing Test HAVE_PROCFS_MAPS - Failed
-- Performing Test HAVE_PROCFS_STAT
CMake Error: TRY_RUN() invoked in cross-compiling mode, please set the following cache variables appropriately:
   HAVE_PROCFS_STAT_EXITCODE (advanced)
For details see /home/sjlee/git/coreclr/bin/obj/Linux.arm64.Debug/TryRunResults.cmake
-- Performing Test HAVE_PROCFS_STAT - Failed
-- Performing Test HAVE_PROCFS_STATUS
CMake Error: TRY_RUN() invoked in cross-compiling mode, please set the following cache variables appropriately:
   HAVE_PROCFS_STATUS_EXITCODE (advanced)
For details see /home/sjlee/git/coreclr/bin/obj/Linux.arm64.Debug/TryRunResults.cmake
-- Performing Test HAVE_PROCFS_STATUS - Failed
-- Performing Test HAVE_COMPATIBLE_ACOS
-- Performing Test HAVE_COMPATIBLE_ACOS - Success
-- Performing Test HAVE_COMPATIBLE_ASIN
-- Performing Test HAVE_COMPATIBLE_ASIN - Success
-- Performing Test HAVE_COMPATIBLE_POW
-- Performing Test HAVE_COMPATIBLE_POW - Failed
```

**Comment by lemmaa** [comment]
@jkotas , PTAL.

**Comment by lemmaa** [comment]
Fixed by #5406
